api/apps/api/src/modules/users/users.service.ts

```diff
@@ -216,4 +221,47 @@ export class UsersService extends AppBaseService<
   private hash(password: string) {
     return hash(password, 10);
   }
+
+  private async isPlatformAdmin(userId: string): Promise<boolean> {
+    return (await this.adminRepo.count({ where: { userId } })) > 0;
+  }
+
+  async getPlatformAdmins(
+    userId: string,
+  ): Promise<Either<typeof forbiddenError, PlatformAdminEntity[]>> {
+    if (!(await this.isPlatformAdmin(userId))) {
+      return left(forbiddenError);
+    }
+    return right(await this.adminRepo.find());
+  }
+
+  async addAdmin(
+    loggedUserId: string,
+    userId: string,
+  ): Promise<Either<typeof forbiddenError, void>> {
+    return this.adminRepo.manager.transaction(async (entityManager) => {
+      if (!(await this.isPlatformAdmin(loggedUserId))) {
+        return left(forbiddenError);
+      }
+
+      const admin = new PlatformAdminEntity();
+      admin.userId = userId;
+
+      await entityManager.save(admin);
```

Comment:
by now we only know (because of `ParseUUIDPipe` that `admin.userId` is a UUID, but we don't know if there's any user with such `id`. we can let this fail (but we'd need to catch and handle the possible exception) or check validate the `userId` first.

personally, I'd let it fail and handle the failure, which _also_ helps in with Murphy's law (a user I am about to make a platform admin deletes their account in the very millisecond between when we check if such user exists and when we try to grant them platform admin role... yeah, as if...)

Reply:
also, why the transaction here? we only have one single `INSERT`, after all.

Reply:
Wouldn't the transaction protect the user from being deleted until this has passed? So when this is triggered, that nanosecond between deletion and addition should not matter because of the transaction?

Reply:
yes, you are right. if we get a request to delete the user by the time we've started this transaction, the user will only be deleted after the transaction is committed 🚀